src/cript/api/api.py

```diff
@@ -511,22 +510,42 @@ def save(self, project: Project) -> None:
 
         Returns
         -------
-        None
+        A set of extra saved node UUIDs.
             Just sends a `POST` or `Patch` request to the API
         """
+        if known_uuid is None:
+            known_uuid = set()
 
-        project.validate()
-
+        node.validate()
         # saves all the local files to cloud storage right before saving the Project node
         # Ensure that all file nodes have uploaded there payload before actual save.
-        for file_node in project.find_children({"node": ["File"]}):
+        for file_node in node.find_children({"node": ["File"]}):
             file_node.ensure_uploaded(api=self)
 
-        response: Dict = requests.post(url=f"{self._host}/{project.node_type.lower()}", headers=self._http_headers, data=project.json).json()
+        node_known = len(self.search(type(node), SearchModes.UUID, str(node.uuid)).current_page_results) == 1
```

Comment:
I wrote a `get_bad_uuid_from_error_message` that I think would be helpful for getting the UUID from the error message

Reply:
yeah, I didn't see or use that function. we can use it here.

Reply:
well, actually no.

This is doing something else then you think.

This just checks if a UUID is known in the backend.
We can use a helper function for that too, if you like.

Reply:
I do like helper functions!

Reply:
but I agree with you, lets get it working and then lets refactor it to make it better